### PR TITLE
Add simple string operations to stdlib

### PR DESCRIPTION
**Pull request by johnwikman**
Adds some string operations to stdlib. Some of these are copied from the stringops.mc example.

### CODE REVIEW

File: stdlib/string.mc [hunk 1]
```diff
@@ -29,6 +31,63 @@ let int2string = fix (lam int2string. lam n.
...
+    if eqi (length s) 0 then
+      if eqi acc (negi 1)
+      then None
+      else Some(acc)
+    else
+      if eqchar c (head s)
+      then strrindex (addi i 1) i   c (tail s)
+      else strrindex (addi i 1) acc c (tail s)
+  ) in
+  strrindex_abs 0 (negi 1) c s
```
Comment (EliasC) [inline stdlib/string.mc]: May I suggest a name that is more than one letter different from `strindex`? Maybe call them `strIndex` and `strLastIndex`?

Comment (EliasC) [reply stdlib/string.mc]: (or non-camelCase, if you prefer. We're going to have to go over the whole standard library at some later point to make naming convention the same all over)

File: stdlib/string.mc [hunk 2]
```diff
@@ -29,6 +31,63 @@ let int2string = fix (lam int2string. lam n.
     concat (int2string (divi n 10)) d
 )
 
+// Returns an option with the index of the first occurrence of c in s. Returns
+// None if c was not found in s.
+let strindex = lam c. lam s.
+  let strindex_abs = fix (lam strindex. lam i. lam c. lam s.
```
Comment (EliasC) [inline stdlib/string.mc]: Minor comment: the `foo_abs` naming convention is something I used in the experiments where I was separating `fix` from definition. It does not "mean" anything in particular.

**Comment by EliasC** [comment]
Looking good in general! Some minor comments regarding naming. Let me know when you have adressed (or refuted!) these comments, and I will merge.

File: stdlib/string.mc (hunk 1 above)
Comment (johnwikman) [reply stdlib/string.mc]: Done. I usually prefer a non-camelCase naming scheme, but it is better to have a consistent naming scheme (I saw that you use camelCase for `zipWith` etc.)

Comment (EliasC) [reply stdlib/string.mc]: Awesome! Merging now...

**Pull request merged**